### ISSUE TITLE
Not able to find vioaltions

### ISSUE THREAD

**Issue by DhirajKhot** [nickcolley/jest-axe#99]
I am writing a sample test for accessibility scan. I could see that the axe run is not able to find the violations in the inner HTML.

Example unit test : 
```
test("Sample Accessibility Test", async () => {
	const sampleHTML: string = "<div class=\"abc1\" role=\"main\"><div class=\"page-content\"><div class=\"abc1-container\"><div class=\"abc1-content\"><div class=\"abc2\"><button type=\"button\" id=\"my-button\" class=\"my-button\" data-is-focusable=\"true\"><span class=\"abc3\" data-automationid=\"splitbuttonprimary\"><span class=\"textContainer\"><span class=\"label\" id=\"my-label\">Sample Button</span></span></span></button></div></div></div></div></div>";
	const results = await axe(sampleHTML);
	expect(results).toHaveNoViolations();
});
```
Actual - 
This test passes with no violations. 

Expected - 
The test should fail with violations related to aira-labels for the button.

**Comment by jrpool** [comment]
I was using jest-axe 3.2.0 in an e2e/src/integration/app.spec.ts file to test an app.component.html file in an Angular 9 project, and jest-axe was behaving as expected. Now I have updated jest-axe to 3.4.0, @types/jest-axe to 3.2.2, and axe-core to 3.5.3, and I can do hardly any damage to the HTML template that causes jest-axe to report a failure. So far I have tried about a dozen blatant faults, and jest-axe ignores them all. I have found only one that jest-axe complains about: giving two elements the same "id" attribute AND ALSO referencing that ID with the "for" attribute of a "label" element.

**Comment by jrpool** [comment]
Correction made above in the path of the Jest spec file.

**Comment by DhirajKhot** [comment]
@nickcolley 
In this case, do we have an issue with jest-axe or the axe-core engine itself?

**Comment by jrpool** [comment]
It is a jest-axe issue. The axe extension of Chrome reports the violations.

**Comment by DhirajKhot** [comment]
@jrpool 
If I try using the axe-core itself using below - 

```
test.only("Sample Home page Accessibilty test", (done) => {
	const sampleHTML: string = "<button type=\"button\" id=\"my-button\" class=\"my-button\" data-is-focusable=\"true\"><span class=\"abc3\" data-automationid=\"splitbuttonprimary\"><span class=\"textContainer\"><span class=\"label\" id=\"my-label\">Sample Button</span></span></span></button>";
	var doc = new DOMParser().parseFromString(sampleHTML, "text/xml");
	Axe.configure({ rules: Axe.getRules(["wcag2a", "wcag2aa", "wcag21a", "wcag21aa", "section508", "best-practice"]) as Axe.Rule[] })
	Axe.run(doc).then(results => {
		console.log("Vioaltions - " + results.violations.length);
		expect(results.violations.length).toBeLessThan(1);
		done();
	});
});
```

Still I get no violations.

**Comment by jrpool** [comment]
Fascinating. I'll try something similar.

**Comment by nickcolley** [comment]
Hey folks, thanks for looking into this.

I'll close this out since it's not have activity but if you find anything that needs fixing in jest-axe feel free to re-open.
